test/base/src/androidTest/kotlin/io/realm/test/shared/RealmSchemaTests.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.shared
+
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.entities.Sample
+import io.realm.entities.schema.SchemaVariations
+import io.realm.schema.ListPropertyType
+import io.realm.schema.RealmPropertyType
+import io.realm.schema.RealmStorageType
+import io.realm.schema.SingularPropertyType
+import io.realm.test.platform.PlatformUtils
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+public class RealmSchemaTests {
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration =
+            RealmConfiguration.Builder(schema = setOf(SchemaVariations::class, Sample::class))
+                .path("$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun realmClass() {
```

Comment:
Missing test where the class isn't part of the schema.

Reply:
Added `realmClass_notFound` and `realmProperty_notFound`